src/main/java/com/code_intelligence/jazzer/mutation/support/ParameterizedTypeSupport.java

```diff
@@ -64,21 +64,50 @@ public class ParameterizedTypeSupport {
    */
   public static AnnotatedType resolveTypeArguments(
       Class<?> clazz, AnnotatedType classType, AnnotatedType type) {
-    if (!(classType instanceof AnnotatedParameterizedType)) {
+    Map<TypeVariable<?>, AnnotatedType> mapping = new HashMap<>();
+    collectTypeMappings(clazz, classType, mapping);
+    if (mapping.isEmpty()) {
       return type;
     }
+    return resolveRecursive(type.getType(), type, mapping);
+  }
 
-    TypeVariable<?>[] typeParameters = clazz.getTypeParameters();
-    AnnotatedType[] typeArguments =
-        ((AnnotatedParameterizedType) classType).getAnnotatedActualTypeArguments();
+  private static void collectTypeMappings(
```

Comment:
Minor: `collect` sounds functional to me! A name hinting at mutation might be better. Maybe `updateTypeMappings`?